R/resultAnnotations.R

```diff
@@ -0,0 +1,925 @@
+#'     default, the blacklist regions as reported in \cite{...} are used.
+#'
+#' @param fds A FraserDataSet
+#' @param txdb A txdb object providing the reference annotation.
+#' @param result A result table as generated by FRASER, including the column 
+#'     \code{annotatedJunction} as generated by the function
+#'     \code{annotateIntronReferenceOverlap}.
+#' @param addSpliceType Logical, indicating if the type of the splice event 
+#'     should be added to the results table. Defaults to \code{TRUE}.
+#' @param addUTRoverlap Logical, indicating if the overlap with UTR regions
+#'     should checked and added to the results table. Defaults to \code{TRUE}.
+#' @param blacklist_regions A BED file that contains the blacklist regions. 
+#'     If \code{NULL} (default), the BED files that are packaged with FRASER 
+#'     are used. 
+#' @param assemblyVersion Indicates the genome assembly version of the intron 
+#'     coordinates. Only used if blacklist_regions is NULL. For other versions, 
+#'     please provide the BED file containing the blacklist regions directly.
+#' @param BPPARAM For controlling parallelization behavior. Defaults to 
+#'     \code{bpparam()}.
+#' @return An annotated FraserDataSet or results table, respectively
+#' 
+#' @name spliceTypeAnnotations
+#' @rdname spliceTypeAnnotations
+#' 
+#' @examples
+#'   # get data, fit and compute p-values and z-scores
+#'   fds <- createTestFraserDataSet()
+#'   
+#'   # load reference annotation
+#'   requireNamespace("TxDb.Hsapiens.UCSC.hg19.knownGene")
+#'   txdb <- 
+#'     TxDb.Hsapiens.UCSC.hg19.knownGene::TxDb.Hsapiens.UCSC.hg19.knownGene 
+#'     
+#'   # add basic annotations for overlap with the reference annotation
+#'   # run this function before creating the results table
+#'   fds <- annotateIntronReferenceOverlap(fds, txdb) 
+#' 
+#'   # extract results: for this small example dataset, only a z score cutoff 
+#'   # of 1 is used to get at least one result. 
+#'   # Make sure to include the additional column in the results table
+#'   res <- results(fds, padjCutoff=NA, zScoreCutoff=1, deltaPsiCutoff=NA, 
+#'                   additionalColumns = 'annotatedJunction')
+#'   res_dt <- as.data.table(res)
+#'   
+#'   # annotate the type of splice event and UTR overlap
+#'   res_dt <- annotateSpliceEventType(result=res_dt, txdb=txdb, fds=fds)
+#'   
+#'   # annotate overlap with blacklist regions
+#'   res_dt <- flagBlacklistRegions(result=res_dt, assemblyVersion="hg19")
+#'   
+#'   # show results table containing additional annotations
+#'   res_dt
+#'   
+NULL
+
+#' @describeIn spliceTypeAnnotations This method calculates basic annotations 
+#'     based on overlap with the reference annotation (start, end, none, both) 
+#'     for the full fds. The overlap type is added as a new column 
+#'     \code{annotatedJunction} in \code{mcols(fds)}. 
+#' @export
+annotateIntronReferenceOverlap <- function(fds, txdb, BPPARAM=bpparam()){
+    print("loading introns")
```

Comment:
```suggestion
    message("loading introns")
```
It is discouraged to use `print` within packages.